governance/program/src/instruction.rs

```diff
@@ -0,0 +1,214 @@
+    ///   6. `[writable]` Initialized Signatory account for the issued signatory token
+    ///   7. `[writable]` Initialized Source Token Holding account
+    ///   8. `[]` Source mint account
+    ///   9. `[]` Proposal Authority account. PDA with seeds: ['governance',proposal_key]
+    ///   10. '[]` Token program account
+    ///   11. `[]` Rent sysvar
+    InitializeProposal {
+        /// Link to gist explaining proposal
+        description_link: [u8; MAX_PROPOSAL_DESCRIPTION_LINK_LENGTH],
+        /// Name of the proposal
+        name: [u8; MAX_PROPOSAL_NAME_LENGTH],
+    },
+
+    /// [Requires Admin token]
+    /// Adds a signatory to the Proposal which means that this Proposal can't leave Draft state until yet another signatory burns
+    /// their signatory token indicating they are satisfied with the Proposal instructions.
+    /// They'll receive an signatory token as a result of this call that they can burn later
+    ///
+    ///   0. `[writable]` Initialized Signatory account
+    ///   1. `[writable]` Initialized Signatory Mint account
+    ///   2. `[writable]` Admin account
+    ///   3. `[writable]` Proposal State account
+    ///   4. `[]` Proposal account
+    ///   5. `[]` Transfer authority
+    ///   6. `[]` Proposal Authority account. PDA with seeds: ['governance',proposal_key]
+    ///   7. '[]` Token program account
+    AddSignatory,
+
+    /// [Requires Admin token]
+    /// Removes a signer from the Proposal
+    ///
+    ///   0. `[writable]` Signatory account to remove token from
+    ///   1. `[writable]` Signatory Mint account
+    ///   2. `[writable]` Admin account
+    ///   3. `[writable]` Proposal State account
+    ///   4. `[]` Proposal account
+    ///   5. `[]` Transfer authority
+    ///   6. `[]` Proposal Authority account. PDA with seeds: ['governance',proposal_key]
+    ///   7. '[]` Token program account
+    RemoveSignatory,
+
+    /// [Requires Signatory token]
+    /// Adds a Transaction to the Proposal Max of 5 of any Transaction type. More than 5 will throw error.
+    /// Creates a PDA using your authority to be used to later execute the instruction.
+    /// This transaction needs to contain authority to execute the program.
+    ///
+    ///   0. `[writable]` Uninitialized Proposal Transaction account
+    ///   1. `[writable]` Proposal state account
+    ///   2. `[writable]` Signatory account
+    ///   3. `[]` Proposal account.
+    ///   4. `[]` Governance account
+    ///   5. `[]` Transfer authority
+    ///   6. `[]` Proposal Authority account. PDA with seeds: ['governance',proposal_key]
+    ///   7. `[]` Governance program account
+    AddCustomSingleSignerTransaction {
+        /// Slot waiting time between vote period ending and this being eligible for execution
+        hold_up_time: u64,
+        /// Instruction
+        instruction: Vec<u8>,
+        /// Position in transaction array
+        position: u8,
```

Comment:
How about just supporting a single instruction?   If necessary, that instruction could be to a program that in turn executes N other instructions.

Reply:
The idea behind supporting multiple instructions is to allow a single proposal for multiple instructions which must be spaced in time. 
For example you can have a proposal for disbursement of founds in 3 yearly instalments. In order to achieve that you can create 3 transfer instructions with hold up time 1, 2 and 3 years accordingly.

Reply:
Ah, I see. Do we actually have a list of requirements for what this program is trying to meet?

Reply:
It was inspired by COMP Governance https://compound.finance/docs/governance